Sources/swift-format/Frontend/Frontend.swift

```diff
@@ -208,4 +211,16 @@ class Frontend {
     // default configuration.
     return Configuration()
   }
+
+  /// Checks if all the rules in the given configuration are supported by the registry.
+  /// If there are any rules that are not supported, they are emitted as a warning.
+  private func verifyConfigurationRules(for configuration: Configuration) -> Void {
+    // If any rules in the decoded configuration are not supported by the registry,
+    // emit them into the diagnosticsEngine as warnings.
+    // That way they will be printed out, but we'll continue execution on the valid rules.
+    let invalidRules = configuration.rules.filter { !RuleRegistry.rules.keys.contains($0.key) }
+    if !invalidRules.isEmpty {
+      diagnosticsEngine.emitWarning("Configuration contains an unsupported rule: \(invalidRules.keys.joined(separator: ", "))", location: nil)
```

Comment:
```suggestion
      diagnosticsEngine.emitWarning("Configuration contains an unrecognized rule: \(invalidRules.keys.joined(separator: ", "))", location: nil)
```

Reply:
"Unrecognized" feels clearer than "Unsupported". In my mind the latter could be interpreted as "this rule exists, but it's not supported for some reason".